Add 15 December 2024 outage post-mortem (#138)
openstreetmap/owg-website: 2025/02/15/post-mortem.md

# Post-Mortem: Network Outage Affecting OpenStreetMap.org - 15 December 2024

> 

Reviewer: ```suggestion

All times are provided in Coordinated Universal Time (UTC).

```

Are only the times with "(UTC)" in UTC, or are "all times in UTC"? (if so, you can remove it elsewhere)

   - OSM Operations team emails HE.net’s Network Operations Centre (NOC), inquiring if there is any unplanned maintenance.

5. **15 December 2024, 04:24 – HE.net NOC Response**
   - HE.net confirms via email that there is no scheduled maintenance. They are investigating the outage.

6. **15 December 2024, 04:43 – HE.net Phone Status**
   - OSM Operations contacts HE.net by telephone.
   - HE.net states the outage is in Amsterdam, with no estimated time to recover. They are waiting for Equinix "remote hands" to investigate.

7. **15 December 2024, 07:24 – Second Request for ETA**
   - OSM Operations again requests an estimated time of recovery from HE.net.
   - No precise estimate provided.

8. **15 December 2024, 11:31 – Read-Only Backup Online**
   - A read-only backup instance of OpenStreetMap.org (hosted in Dublin) is activated, providing partial functionality.
   - A maintenance notice is displayed on OpenStreetMap.org, indicating that data edits cannot be saved.

9. **15 December 2024, 11:12 & Ongoing – Additional Enquiries**
   - OSM Operations team continues to request updates from HE.net at regular intervals, but no ETA is provided.

> 10. **16 December 2024, TBC - Operations Team Prepare Emergency Full Failover Procedure Documention**

Reviewer 2: It was about 10am when I wrote my plan - last edit was 10:11 according to hackmd.

Author: Thanks. Fixed.

> 10. **16 December 2024, 10:11 - Operations Team Prepare Emergency Full Failover Procedure Documention**

Reviewer: ```suggestion
10. **16 December 2024, 10:11 - Operations Team Prepare Emergency Full Failover Procedure Documentation**
```

Should be "Documentation" (typo)

>     - Document the required steps/procedure to fully failover to Dublin (Postgres & osmdbt)

Reviewer: ```suggestion
    - The required steps/procedure to fully failover to Dublin (Postgres & osmdbt) are documented.
```

Continue using 'present' tense (inconsistent)

> 11. **16 December 2024, TBC - Operations Team Decision to postpone the Emergency Full Failover**

Reviewer 2: IRC logs say it was 11:55 when we made that decision.

Author: Thanks. Fixed.

    - OSM Operations team decide to wait for restoration of Amsterdam connectivity instead of Emergency Full Failover.

>     - Communicate the expected ETA to full restoration of services with the OpenStreetMap Communication Working Group, OSMF Board and wider OpenStreetMap community (Announcement Mailinglist, Talk Mailing List, [Mastodon](https://en.osm.town/@osm_tech), [Community Forum](https://community.openstreetmap.org/))

Reviewer: ```suggestion
    - The expected ETA to full restoration of services is communicated with the OpenStreetMap Communication Working Group, OSMF Board and wider OpenStreetMap community (Announcement Mailinglist, Talk Mailing List, [Mastodon](https://en.osm.town/@osm_tech), [Community Forum](https://community.openstreetmap.org/)).
```

Same as above

12. **16 December 2024 - Escalation of Equinix Internet connectivity installations.**

>     - Equinix Teams kindly escalated the installations Equinix Internet connectivity in Amsterdam and Dublin.

Reviewer: ```suggestion
    - Equinix Teams kindly escalates the Equinix Internet connectivity installations in Amsterdam and Dublin.
```

Same as above

- **Contributing Factors:**
  - Lack of a fully operational second ISP at the time of the outage.
  - Delayed resolution due to HE.net needing to ship replacement hardware from the United States, extending the downtime.
  - Inability to fully failover to secondary site due to incomplete replication and operational complexity & perceived risk.

---

## 5. Resolution and Mitigation Steps

1. **Short-Term Resolution:**
   - Deployed a read-only backup instance in Dublin to provide partial functionality for the main website.
   - Implemented a new Equinix Internet ISP link on 17 December 2024, restoring full functionality prior to HE.net fixing their equipment.

2. **Long-Term Mitigation:**
   - Transitioned to a multi-ISP architecture to ensure that, if one provider fails, another can switch over with minimal interruption.
   - Improved out-of-band connection, ensuring that connection is reliable and can be used for data recovery to secondary site.

---

## 6. Lessons Learned

1. **Single Point of Failure (SPOF) and Redundancy**
   - Relying on HE.net alone exposed the Amsterdam servers to a single point of failure.

>    - In future, we will maintain multiple upstream providers to eliminate critical SPOFs.

Reviewer: ```suggestion
   - In the future, we will maintain multiple upstream providers to eliminate critical SPOFs.
```

Change grammar (typo)

4. **Review and Update ISP SLAs**
   - **Description:** Engage with HE.net and Equinix Internet on contractual SLAs and incorporate defined escalation procedures and response times.

5. **Improve Out-of-Band Access**
   - **Description:** The Raspberry Pi 4 and 4G modem used for out-of-band access. The out-of-band system should be pre-configured to be able to be used as a manual fallback uplink in the event of an ISP outage. The link would be used for server access and syncing the Postgres replication data to the secondary site. The out-of-band hardware should be upgraded to ensure resilience and redundancy. https://github.com/openstreetmap/operations/issues/1197

6. **Document Outage Communication Procedures**
   - **Description:** A document should be created to outline the parties and their responsibilities in communicating outages. The document should list the recommended communication channels.

7. **Investigate possible to change OpenStreetMap.org Authentication when Database is Read-Only**

>    - **Description:** The OpenStreetMap.org "rails-port" maintainers should investigate if it is possible and practical to de-couple the OAuth authentication to allow the authentication to continue to function for third parties during times when the site is disrupted.

Reviewer: ```suggestion
   - **Description:** The OpenStreetMap.org "rails-port" maintainers should investigate if it is possible and practical to de-couple the OAuth service to allow authentication to continue to function for third parties during times when the site is disrupted.
```
Changes grammar (remove redundant wording)

---

## 8. Conclusion

This outage emphasised the importance of robust network redundancy and vendor reliability. By investing in multiple ISPs, improving monitoring and alerting, and refining both backup and communication procedures, OpenStreetMap can continue to provide a highly available service to our global community. The outlined action items will help to address the weaknesses exposed by this incident and strengthen OpenStreetMap.org against similar disruptions in the future.

> ## 9. Thank you to all those that helped

Reviewer: ```suggestion
## 9. Thank you
```

Use short headings (legibility)

> We very generously received support and assistance from many people around the world.

Reviewer: ```suggestion
Thank you to all those that helped. We very generously received support and assistance from many people around the world.
```

Use topic sentence (legibility)
